### PR TITLE
Added VA test

### PR DESCRIPTION
**Pull request by PraveenPenguin**
added VA test which test virtual address limit on  linux

Signed-off-by: Praveen K Pandey <praveen@linux.vnet.ibm.com>

### CODE REVIEW

File: memory/vatest.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        # Check for basic utilities
+        smm = SoftwareManager()
+        for packages in ['gcc', 'make']:
+            if not smm.check_installed(packages) and not smm.install(packages):
+                self.cancle('%s is needed for the test to be run' % packages)
+
+        data_dir = os.path.abspath(self.datadir)
+
+        shutil.copyfile(os.path.join(data_dir, 'va_test.c'),
+                        os.path.join(self.srcdir, 'va_test.c'))
```
Comment (narasimhan-v) [inline memory/vatest.py]: https://github.com/avocado-framework/avocado/issues/1924

With `srcdir` being agreed upon for deprecation, please use an alternative

File: memory/vatest.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                        os.path.join(self.srcdir, 'Makefile'))
+
+        build.make(self.srcdir)
+
+    def test(self):
+        '''
+        Execute VA test
+        '''
+        os.chdir(self.srcdir)
+        scenario_arg = self.params.get('scenario_arg', default=1)
```
Comment (narasimhan-v) [inline memory/vatest.py]: better to have in `setUp()`

File: memory/vatest.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        if scenario_arg in [2, 3, 4]:
+            if memory.meminfo.Hugepagesize.mb != 16:
+                self.cancel(
+                    "Test need to skip as 16MB huge need to configured")
+            if scenario_arg == 4:
+                memory.set_num_huge_pages(131072)
+            else:
+                memory.set_num_huge_pages(1024)
+        elif scenario_arg in [5, 6, 7]:
+            if memory.meminfo.Hugepagesize.gb != 16:
```
Comment (narasimhan-v) [inline memory/vatest.py]: can this check be done before the `scenario_arg` check ?

File: memory/vatest.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    def setUp(self):
+        '''
+        Build VA Test
+        '''
+
+        # Check for basic utilities
+        smm = SoftwareManager()
+        for packages in ['gcc', 'make']:
+            if not smm.check_installed(packages) and not smm.install(packages):
+                self.cancle('%s is needed for the test to be run' % packages)
```
Comment (narasimhan-v) [inline memory/vatest.py]: cancel

File: memory/vatest.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+        data_dir = os.path.abspath(self.datadir)
```
Comment (narasimhan-v) [inline memory/vatest.py]: is this extra variable needed ? I guess `self.datadir` already holds the absolute path

File: memory/vatest.py (hunk 2 above)
Comment (narasimhan-v) [reply memory/vatest.py]: Also the below check can be in `setUp()`

File: memory/vatest.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                self.cancel("Test need to skip as 16GB huge need to configured")
+            if scenario_arg == 7:
+                memory.set_num_huge_pages(2)
+            else:
+                memory.set_num_huge_pages(1)
+
+        result = process.run('./va_test -s %s' %
+                             scenario_arg, shell=True, ignore_status=True)
+        for line in result.stdout.splitlines():
+            if 'Problem' in line:
```
Comment (narasimhan-v) [inline memory/vatest.py]: Should we not check all the `Problem`s instead of failing fast ?

File: memory/vatest.py (hunk 4 above)
Comment (PraveenPenguin) [reply memory/vatest.py]: ah sure typo , will rectify

File: memory/vatest.py (hunk 5 above)
Comment (PraveenPenguin) [reply memory/vatest.py]: i will remove this

File: memory/vatest.py (hunk 1 above)
Comment (PraveenPenguin) [reply memory/vatest.py]: sure i will

File: memory/vatest.py (hunk 2 above)
Comment (PraveenPenguin) [reply memory/vatest.py]: will move this on setup

File: memory/vatest.py (hunk 3 above)
Comment (PraveenPenguin) [reply memory/vatest.py]: that is requirement as only  scenario 5, 6 , 7 need 16 GB as huge page configured

File: memory/vatest.py (hunk 6 above)
Comment (PraveenPenguin) [reply memory/vatest.py]: should be fine as test code file wrote like that way if after first problem execution if test code exit (c file ) as another occurrence is not possible , any way will check this and try to modify test code (c file )

**Comment by PraveenPenguin** [comment]
@narasimhan-v thanks for review , raised new PR #683  with suggested change please have a look

File: memory/vatest.py (hunk 3 above)
Comment (narasimhan-v) [reply memory/vatest.py]: This check is done for both 2,3,4 and 5,6,7

Comment (PraveenPenguin) [reply memory/vatest.py]: no i don't think so for  scenario 2,3,4 we need 16MB huge page should configured

Comment (narasimhan-v) [reply memory/vatest.py]: What I ask is, you are checking `memory.meminfo.Hugepagesize.mb != 16` for both cases. If that is needed, we can handle the check together

**Pull request closed without merging**